conf/parse.go

```diff
@@ -98,13 +102,60 @@ func ParseFileWithChecks(fp string) (map[string]interface{}, error) {
 	return p.mapping, nil
 }
 
+// cleanupUsedEnvVars will recursively remove all already used
+// environment variables which might be in the parsed tree.
+func cleanupUsedEnvVars(m map[string]interface{}) {
+	for k, v := range m {
+		t := v.(*token)
+		if t.usedVariable {
+			delete(m, k)
+			continue
+		}
+		// Cleanup any other env var that is still in the map.
+		if tm, ok := t.value.(map[string]interface{}); ok {
+			cleanupUsedEnvVars(tm)
+		}
+	}
+}
+
+// ParseWithDigest returns the processed config and a digest
+// that represents the configuration.
+func ParseWithDigest(fp string) (map[string]interface{}, string, error) {
```

Comment:
Minor thing but let's make this naming clearer based on the other functions it is replacing, i.e. `ParseFileWithChecksDigest`.